src/bloom/hierarchical.rs

```diff
@@ -0,0 +1,335 @@
+use std::iter::FromIterator;
+
+use super::*;
+
+/// Container for types which is support bloom filtering
+#[derive(Debug)]
+pub struct HierarchicalBloom<Child>
+where
+    Child: BloomProvider + Send + Sync,
```

Comment:
If you can, you should try to avoid trait constraints on generic variables in struct definitions and only add them to impl clauses (as it's easier to add/remove new constraints later on)